langchain/src/text_splitter.ts

```diff
@@ -216,7 +221,10 @@ export class CharacterTextSplitter
   async splitText(text: string): Promise<string[]> {
     // First we naively split the large input into a bunch of smaller ones.
     const splits = this.splitOnSeparator(text, this.separator);
-    return this.mergeSplits(splits, this.keepSeparator ? "" : this.separator);
+    return await this.mergeSplits(
```

Comment:
Don't need an `await` here

Reply:
@jacoblee93 Resolved it, thanks! The use case is for using getNumTokens as a custom length function, just as can be done in langchain python. So any text splitter can allow for using token counting instead of character counting. 

This is  important in applications such as chunking for vector retrieval, particularly when you're working with multiple languages. The openAI tokenizer is optimized for different languages differently, and some languages uses way more tokens per character than seen with English language tokenization. So if you want to retrieve, say 8 chunks, and you set that so you don't go over context size limits, then using a RecursiveTextSplitter using just characters gets tricky. Instead of changing the number of chunks retrieved by the retrieved, you can enforce a token limit when chunking using any splitter.



Eg:
```
  const llmModel = new OpenAI({
    openAIApiKey: process.env.OPENAI_API_KEY,
    modelName: 'gpt-3.5-turbo',
  });

  const tokenLengthFunction = llmModel.getNumTokens

  const textSplitter = new RecursiveCharacterTextSplitter({
     chunkSize: 500, // Count using tokens
     chunkOverlap: 0,
     lengthFunction: tokenLengthFunction,  // use the LLM's token length function to count tokens instead of characters
   });
```

Reply:
Gotcha - and some of those functions are async at the moment or is this future-proofing?